src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -276,4 +278,59 @@ public ResponseEntity<Result> createResultForExternalSubmission(@PathVariable Lo
         return ResponseEntity.created(new URI("/api/results/" + savedResult.getId()))
                 .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, savedResult.getId().toString())).body(savedResult);
     }
+
+    /**
+     * GET /exercises/:exerciseId/feedback-details : get all feedback details and participations for an exercise.
+     *
+     * @param exerciseId The ID of the exercise
+     * @return A response entity containing feedback details and participations
+     */
+    @GetMapping("exercises/{exerciseId}/feedback-details")
+    @EnforceAtLeastTutor
+    public ResponseEntity<FeedbackDetailsResponse> getAllFeedbackDetailsForExercise(@PathVariable Long exerciseId) {
+        log.debug("REST request to get all Feedback details for Exercise {}", exerciseId);
+        Exercise exercise = exerciseRepository.findByIdElseThrow(exerciseId);
+
+        // Get all Student participations for the exercise
+        Set<StudentParticipation> participation;
```

Comment:
We have multiple participations, please use plural

```suggestion
        Set<StudentParticipation> participations;
```